File: src/main/resources/output/IFC4X3_python/Entities/IfcProfileDef/IfcParameterizedProfileDef/IfcTShapeProfileDef/IfcTShapeProfileDef.py
class IfcTShapeProfileDef(IfcParameterizedProfileDef):
    def __init__(self):
        self.Depth: IfcPositiveLengthMeasure = None
        self.FlangeWidth: IfcPositiveLengthMeasure = None
        self.WebThickness: IfcPositiveLengthMeasure = None
        self.FlangeThickness: IfcPositiveLengthMeasure = None
        self.FilletRadius: IfcNonNegativeLengthMeasure = None
        self.FlangeEdgeRadius: IfcNonNegativeLengthMeasure = None
        self.WebEdgeRadius: IfcNonNegativeLengthMeasure = None
        self.WebSlope: IfcPlaneAngleMeasure = None
        self.FlangeSlope: IfcPlaneAngleMeasure = None

    # === WHERE CLAUSES ===
    # ValidFlangeThickness : FlangeThickness < Depth
    # ValidWebThickness : WebThickness < FlangeWidth

File: src/main/resources/output/IFC4X3_java/Entities/IfcProfileDef/IfcParameterizedProfileDef/IfcTShapeProfileDef/IfcTShapeProfileDef.java
public class IfcTShapeProfileDef extends IfcParameterizedProfileDef {
    public IfcPositiveLengthMeasure Depth;
    public IfcPositiveLengthMeasure FlangeWidth;
    public IfcPositiveLengthMeasure WebThickness;
    public IfcPositiveLengthMeasure FlangeThickness;
    public IfcNonNegativeLengthMeasure FilletRadius;
    public IfcNonNegativeLengthMeasure FlangeEdgeRadius;
    public IfcNonNegativeLengthMeasure WebEdgeRadius;
    public IfcPlaneAngleMeasure WebSlope;
    public IfcPlaneAngleMeasure FlangeSlope;

    // === WHERE CLAUSES ===
    // ValidFlangeThickness : FlangeThickness < Depth
    // ValidWebThickness : WebThickness < FlangeWidth
}
